### PR TITLE
Integrate RichCodeNavigation targets to help with LSIF generation

### PR DESCRIPTION
**Pull request by BertanAygun**
These targets will be enabled when the environment variable for LSIF generation is set in certain builds, and they generate log files targeted to help index generation. In normal builds they should be no-ops.

### CODE REVIEW

File: eng/targets/Settings.props [hunk 1]
```diff
@@ -132,6 +132,13 @@
     <PackageReference Include="Microsoft.CodeAnalysis.PerformanceSensitiveAnalyzers" Version="$(MicrosoftCodeAnalysisPerformanceSensitiveAnalyzersVersion)" PrivateAssets="all" />
   </ItemGroup>
 
+  <!-- 
+     Code indexing targets to help generating LSIF from indexing builds.
+  -->
+  <ItemGroup>
+    <PackageReference Include="RichCodeNav.EnvVarDump" Version="0.1.1407-alpha" />
```
Comment (jasonmalinowski) [inline eng/targets/Settings.props:139]: This should probably be defining a version variable for the version number like we are doing for the packages above. The variable goes in eng/Versions.props.

Comment (BertanAygun) [reply eng/targets/Settings.props]: Fixed this in the 2nd commit.

**Comment by BertanAygun** [comment]
@jasonmalinowski, can you help me rerun CI build? Looks like the second one failed due to an infrastructure issue (timed out) while first one was fine.

**Comment by jasonmalinowski** [comment]
@BertanAygun Reran, but now there's a merge conflict so you'll have to resolve that anyways.

**Pull request merged**